code/Test_definitions/device-reachability-status.feature

```diff
@@ -0,0 +1,149 @@
+    Given the resource "{api-root}/device-reachability-status/v0.6/retrieve" set as base-url                                                             |
+    And the header "Content-Type" is set to "application/json"
+    And the header "Authorization" is set to a valid access token
+    And the header "x-correlator" is set to a UUID value
+	
+############# Happy Path Scenarios ##################	
+
+  @device_reachability_status_01_reachabilityStatusConnectedSms
+  Scenario: Check the reachability status if device is connected with SMS
+    Given a valid devicestatus request body 
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    When the request "getReachabilityStatus" is sent
+    Then the response code is 200
+    And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response body complies with the OAS schema at "/components/schemas/ReachabilityStatusResponse"
+    And the response property "$.status" is 200
+    And the response property "$.code" is "OK"
+    And the response property "$.message" contains a user friendly text
+    And the response property "$.reachabilityStatus" is "CONNECTED_SMS"
+
+  @device_reachability_status_02_reachabilityStatusConnectedData
+  Scenario: Check the reachability status if device is connected with DATA  
+    Given a valid devicestatus request body 
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    When the request "getReachabilityStatus" is sent
+    Then the response code is 200
+    And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response body complies with the OAS schema at "/components/schemas/ReachabilityStatusResponse"
+    And the response property "$.status" is 200
+    And the response property "$.code" is "OK"
+    And the response property "$.reachabilityStatus" is  "CONNECTED_DATA"
+	
+  @device_reachability_status_03_reachabilityStatusDeviceNotConnected
+  Scenario: Check the reachability status for  DeviceNotConnected
+    Given a valid devicestatus request body 
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    When the request "getReachabilityStatus" is sent
+    Then the response code is 200
+    And the response header "Content-Type" is "application/json"
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response body complies with the OAS schema at "/components/schemas/ReachabilityStatusResponse"
+    And the response property "$.status" is 200
+    And the response property "$.code" is "OK"
+    And the response property "$.reachabilityStatus" is  "NOT_CONNECTED"
+	
```

Comment:
We are sending the same request and expect once it returns SMS, DATA or not connected. I think it should be in the precondition, whether the device is connected to SMS DATA or not connected.